### PR TITLE
AS-452: Add code coverage: Jacoco and Codacy

### PR DESCRIPTION
**Pull request by ansingh7115**
Here's an example of coverage reports being posted to Codacy: https://app.codacy.com/gh/DataBiosphere/terra-workspace-manager/file/54386654192/coverage?bid=22641934&fileBranchId=22641934

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -126,6 +126,15 @@ jobs:
...
 
+    - name: Codacy Coverage Reporter
+      if: matrix.gradleTask == 'unitTest'
+      uses: codacy/codacy-coverage-reporter-action@0.2.0
+      env:
+        CODACY_API_BASE_URL: https://app.codacy.com/gh/DataBiosphere/terra-workspace-manager
+      with:
+        project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+        coverage-reports: build/reports/jacoco/test/jacoco.xml
+
```
Comment (davidangb) [inline .github/workflows/test.yml:137]: I'm not sure what the default behavior is ... if the codacy upload fails, will the entire action fail? Do we want it to? (pro: we notice the problem; con: could interrupt development). Maybe this wants continue-on-error?

**Review by davidangb — APPROVED** [review]
one optional question inline, else :+1:

**Comment by ungwudik** [comment]
> Here's an example of coverage reports being posted to Codacy: https://app.codacy.com/gh/DataBiosphere/terra-workspace-manager/file/54386654192/coverage?bid=22641934&fileBranchId=22641934

I'm getting an `Oops! This file no longer exist in the lastest commit of this branch.` error when I open this example link. Probably not blocking this PR? Just want to let you know

**Review by ungwudik — APPROVED** [review]

**Comment by ansingh7115** [comment]
Oh! It renews every time a coverage report is generated (which is every time the test runs). Here's the new one - https://app.codacy.com/gh/DataBiosphere/terra-workspace-manager/file/54388584175/coverage?bid=22641934&fileBranchId=22641934

**Comment by ungwudik** [comment]
> Oh! It renews every time a coverage report is generated (which is every time the test runs). Here's the new one - https://app.codacy.com/gh/DataBiosphere/terra-workspace-manager/file/54388584175/coverage?bid=22641934&fileBranchId=22641934

Ah coool!

**Pull request merged**